### DISCUSSION TITLE
Documentation 4.18. Automatically putting clients on specific workspaces does not work for chromium, vivaldi and spotify

### DISCUSSION THREAD

**Discussion by benibilme** [i3/i3#5262] [Q&A] [answered]
Hello,

I am using arch linux with i3 on pure X environment.  I have following statements in my configuration file. Chromium, vivaldi and spotify do not obey the workspace command. They always pop-up at workspace 1.  Others open up on the allocated workspaces defined below. I do not want to use for_windows command which captures all chromium or vivaldi instances. What is wrong with chromium , vivaldi and spotify. I know that vivaldi is using chrome kernel, I believe their problem is the same.  Spotify may be subject to same problem.  

Any help is much appreciated. 

````
# Start default applications
exec --no-startup-id i3-msg 'workspace 22; exec keepassxc' 
exec --no-startup-id i3-msg 'workspace 21; exec zotero' 
exec --no-startup-id i3-msg 'workspace 20; exec keybase-gui' 
exec --no-startup-id i3-msg 'workspace 20; exec discord' 
exec --no-startup-id i3-msg 'workspace 19; exec chromium https://mail.google.com/mail' 
exec --no-startup-id i3-msg 'workspace 18; exec spotify'   
exec --no-startup-id i3-msg 'workspace 17; exec vivaldi-stable wwww.youtube.com'   
workspace 1
````

**Comment by Airblader** [comment]
This only works if the application supports startup notifications. It's possible the applications in question don't.

**Comment by benibilme** [comment]
After some digging, trial and error and many attempts, following worked for chromium and vivaldi.  According to xprob, tabs are reported as instance. Giving app option fills the instance filed which satisfies my need. 

However, spotify absolutely does not obey any commands. It just pops up on the last focused workspace.  I will dig more for spotify. 

````
assign [class="Chromium" instance="mail.google.com"] $ws20
assign [class="Vivaldi-stable" instance="www.youtube.com"] $ws18 

exec --no-startup-id i3-msg 'workspace $ws18; exec /opt/vivaldi/vivaldi --profile-directory=Default --app=https://www.youtube.com; workspace $ws1' 
exec --no-startup-id i3-msg 'workspace $ws20; exec chromium --profile-directory=Default --app=https://mail.google.com; workspace $ws1'
````

**Reply by Airblader** [reply, marked as answer]
Spotify is known to set its WM_CLASS after mapping it's window, which violates the specification, and causes assign to not work. You need to use for_window here.

**Reply by benibilme** [reply]
You are right. After some digging, I found a reddit post which worked for me.

````
for_window [class="Spotify"] move window to workspace number $ws19
````
